qrisk/stats.py

```diff
@@ -353,15 +431,22 @@ def sortino_ratio(returns, required_return=0, period=DAILY):
         Annualized Sortino ratio.
 
     """
-    try:
-        ann_factor = ANNUALIZATION_FACTORS[period]
-    except KeyError:
-        raise ValueError(
-            "period cannot be: '{}'."
-            " Must be '{}', '{}', or '{}'".format(
-                period, DAILY, WEEKLY, MONTHLY
+
+    if len(returns) < 2:
+        return np.nan
+
+    if annualization is None:
```

Comment:
also, I think that Thomas was suggesting that we pull this try/except out into a function since many functions have the same duplicated code.